integration-tests/src/test/java/oracle/weblogic/kubernetes/ItStickySession.java

```diff
@@ -202,12 +203,41 @@ public void testSameSessionStickinessUsingVoyager() {
     final String ingressName = domainUid + "-ingress-host-routing";
     final String ingressServiceName = VOYAGER_CHART_NAME + "-" + ingressName;
     final String channelName = "tcp-80";
+    final int maxRetry = 60;
 
     // create Voyager ingress resource
     Map<String, Integer> clusterNameMsPortMap = new HashMap<>();
     clusterNameMsPortMap.put(clusterName, managedServerPort);
-    List<String>  hostNames =
-        installVoyagerIngressAndVerify(domainUid, domainNamespace, ingressName, clusterNameMsPortMap);
+
+    List<String>  hostNames = null;
+
+    for (int i = 0; i < maxRetry; i++) {
+      hostNames =
+          installVoyagerIngressAndVerify(domainUid, domainNamespace, ingressName, clusterNameMsPortMap);
+
+      if (hostNames != null && !hostNames.isEmpty()) {
+        break;
+      }
+
+      try {
+        // sometimes the ingress may not be ready even the condition check is ready, sleep a little bit
+        Thread.sleep(1000);
```

Comment:
Why are you using a for-loop and sleep pattern here instead of just using the Awaitility pattern used elsewhere?

Reply:
we are using Awaitility in installVoyagerIngressAndVerify in CommonTestUtils.java already. We just wanted to wait a little more time for the ingress to be created just for this test case without affecting all others tests, plus we need a "Callable<Boolean>" return type to use  Awaitility (I may be wrong here) but installVoyagerIngressAndVerify returns List<String>. That's why Vanaja and I chose to use a for loop here :(